Remove `requests` code

omkar-334 opened the pull request:
# What does this PR do?
Removes `requests` code from setup, src and utils files. 
Fixes #42817 partially

- [x] **Quality checks**: `make fixup` passes with no errors

cc @CoderTCY @Wauplin @Rocketknight1 

notes - 
1. I've removed `requests` from setup/workflow files as well, you might want to check the first commit. 
2. A few formatting fixes like "spacing around operators" got added in as well

omkar-334 committed:
setup, workflow and utils files

omkar-334 committed:
src core files

omkar-334 committed:
Merge branch 'main' into remove-requests-code

Wauplin commented:
Agree with @guibruand 's suggestion here! It's best to use `huggingface_hub.get_session` and then deal with the returned Client as if it was `httpx` directly. @omkar-334 would you like to give it a try? :pray:

Wauplin commented:
Thank you very much! :hugs:

omkar-334 committed:
use get_session instead of httpx

omkar-334 committed:
Merge branch 'main' into remove-requests-code

omkar-334 committed:
fix code quality - remove unused imports

Wauplin commented on src/transformers/pipelines/image_to_image.py:
```diff
@@ -52,7 +54,7 @@ class ImageToImagePipeline(Pipeline):
 
     >>> upscaler = pipeline("image-to-image", model="caidas/swin2SR-classical-sr-x2-64")
-    >>> img = Image.open(io.BytesIO(httpx.get("http://images.cocodataset.org/val2017/000000039769.jpg").content))
+    >>> img = Image.open(io.BytesIO(session.get("http://images.cocodataset.org/val2017/000000039769.jpg").content))
```
also keep `httpx` in examples

Wauplin commented on src/transformers/pipelines/zero_shot_audio_classification.py:
```diff
@@ -107,10 +110,9 @@ def preprocess(self, audio, candidate_labels=None, hypothesis_template="This is
                 # We need to actually check for a real protocol, otherwise it's impossible to use a local file
                 # like http_huggingface_co.png
-                audio = httpx.get(audio, follow_redirects=True).content
+                audio = session.get(audio, follow_redirects=True).content
```
(in general keep `httpx` when previously used)

Wauplin reviewed (commented):
Hi @omkar-334 , thanks for the changes and sorry for the delay before reviewing. I'm very sorry about my previous comment regarding `get_session`. I should have more carefully double-checked the changes and context before answering. It turns out most (if not all) changes made in the PR would better benefit from using `httpx` directly for simplicity rather than `huggingface_hub.get_session`.

While `get_session` is better suited to make requests to the Hub (it has HF-specific features), plain `httpx` is better suited for generic cases that are not HF-related. Also, let's use `httpx` in all 1-file scripts and utilities (since they are mostly meant for devs/maintainers rather than end users).

Sorry again about this change of direction 🙏

omkar-334 committed:
Merge branch 'main' into remove-requests-code

omkar-334 committed:
utils and prevous-http-used and cli

omkar-334 committed:
style changes

omkar-334 committed:
use httpx for non-HF images

omkar-334 committed:
get_session for HF images

omkar-334 committed:
Merge branch 'main' into remove-requests-code

Wauplin committed:
Merge branch 'main' into remove-requests-code

omkar-334 committed:
Merge branch 'main' into remove-requests-code

omkar-334 committed:
change docstrings

omkar-334 committed:
Merge branch 'remove-requests-code' of github-personal:omkar-334/transformers into remove-requests-code

omkar-334 committed:
Merge branch 'main' into remove-requests-code

omkar-334 committed:
revert formatting - 1

omkar-334 committed:
Merge branch 'remove-requests-code' of github-personal:omkar-334/transformers into remove-requests-code

omkar-334 committed:
revert formatting - 2

omkar-334 committed:
docstrings fixes

omkar-334 committed:
revert formatting - 3

omkar-334 committed:
docstring fixes

omkar-334 committed:
fix failing test

omkar-334 commented:
hey @Wauplin , i've reverted the formatting changes and updated the docstrings. I think it's good to go now. 
For some reason, the tests are failing. Could you tell me why, so that i can fix them...

Thanks!

Wauplin commented on src/transformers/testing_utils.py:
```diff
@@ -4115,7 +4115,7 @@ class ColoredFormatter(logging.Formatter):
 
     # Loggers that should be dimmed (less important/verbose)
-    DIMMED_LOGGERS = {"httpx", "httpcore", "urllib3", "requests"}
+    DIMMED_LOGGERS = {"httpx", "httpcore", "urllib3"}
```
```suggestion
    DIMMED_LOGGERS = {"httpx", "httpcore", "urllib3", "requests"}
```

(let's keep it as it might still be used by a subpackage)

Wauplin committed:
Update src/transformers/pipelines/image_to_image.py

Wauplin committed:
Update src/transformers/testing_utils.py

omkar-334 committed:
Merge branch 'main' into remove-requests-code

Wauplin committed:
fix modular generation check

Wauplin committed:
Merge branch 'remove-requests-code' of github.com:omkar-334/transformers into remove-requests-code

Wauplin committed:
fix consistency

Wauplin committed:
Apply suggestions from code review

Wauplin reviewed (approved):
Thanks for this last iteration @omkar-334! I've made a few changes to fix the last remaining issues and reviewed all files manually. Took some time but definitely a good thing to clean up!

Let's wait for a last approval from an official @huggingface/transformers-core-maintainers 🤗

Wauplin committed:
Update src/transformers/models/idefics2/modeling_idefics2.py

Wauplin committed:
Update src/transformers/models/idefics3/modeling_idefics3.py

Wauplin committed:
Merge branch 'main' into remove-requests-code

Wauplin commented:
failing test seems unrelated 

```
FAILED tests/models/udop/test_processing_udop.py::UdopProcessorTest::test_model_input_names - TypeError: TextInputSequence must be str
FAILED tests/models/udop/test_processing_udop.py::UdopProcessorTest::test_processor_with_multiple_inputs - TypeError: TextInputSequence must be str
============ 2 failed, 225 passed, 297 skipped in 108.35s (0:01:48) ============

Exited with code exit status 1
```

I cannot reproduce it locally :confused:

Wauplin committed:
Merge branch 'main' into remove-requests-code

Pediboi666 reviewed (changes requested):
Fertig

Pediboi666 reviewed (changes requested):
Fett
